File: programmers/쿼드압축후_세기.py
zero=0
one=0
def solution(arr):
    h=len(arr)

    def recursion(r, c, s):
        global zero
        global one

        acc=0
        for i in range(r, r+s):
            for j in range(c, c+s):
                acc+=arr[i][j]

        if acc==s*s or acc==0:
            if acc>0:
                one+=1
            else:
                zero+=1
        else:
            recursion(r, c, s//2)
            recursion(r+s//2, c, s//2)
            recursion(r, c+s//2, s//2)
            recursion(r+s//2, c+s//2, s//2)
    recursion(0, 0, h)
    return [zero, one]

File: programmers/쿼드압축후세기.js
function solution (arr) {
  const answer = [0, 0];
  function recursion (len, x, y) {
    let sum = 0;

    for (let i = x; i < x + len; i++) {
      for (let j = y; j < y + len; j++) {
        sum += arr[i][j];
      }
    }

    if (sum === len ** 2 || sum === 0) {
      if (sum === len ** 2) {
        answer[1] += 1;
      } else {
        answer[0] += 1;
      }
    } else {
      const newLen = Math.floor(len / 2);
      recursion(newLen, x, y);
      recursion(newLen, x + newLen, y);
      recursion(newLen, x, y + newLen);
      recursion(newLen, x + newLen, y + newLen);
    }
  }
  recursion(arr.length, 0, 0);

  return answer;
};
